test/python/synthesis/test_qsd_synthesis.py

```diff
@@ -355,7 +359,7 @@ def test_mc_1qubit_opt(self, num_qubits, base_gate):
         qc2 = qsd.qs_decomposition(hidden_mat)
         cqc2 = transpile(qc2, basis_gates=["u", "cx"], optimization_level=0)
         op2 = Operator(qc2)
-        self.assertEqual(hidden_op, op2)
+        self.assertTrue(matrix_equal(hidden_op.to_matrix(), op2.to_matrix(), atol=1e-8))
```

Comment:
What was the tolerance on this check before moving to rust? I'm wondering how loose we're making this now compared to where we were before.